catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloader.java

```diff
@@ -163,16 +163,15 @@ public ResourceResponse setupDownload(Metacard metacard,
         if (downloaderConfig.isCacheEnabled()) {
 
             CacheKey keyMaker = null;
-
+            String key = null;
             try {
                 keyMaker = new CacheKey(metacard, resourceResponse.getRequest());
+                key = keyMaker.generateKey();
```

Comment:
If `new CacheKey()` throws `IllegalArgumentException`, `key` will be `null` which may cause other bad exception later on. Shouldn't we throw an exception or return from the `catch` block instead of pressing forward?